scss/components/_off-canvas.scss

```diff
@@ -6,176 +6,298 @@
-/// @type Shadow
-$maincontent-shadow: 0 0 10px rgba($black, 0.5) !default;
-
 /// Adds baseline styles for off-canvas. This CSS is required to make the other pieces work.
 @mixin off-canvas-basics {
-  // Extra properties needed on <html> and <body> to make off-canvas work
-  html,
-  body {
-    height: 100%;
-  }
-
-  .off-canvas-wrapper {
-    width: 100%;
-    overflow-x: hidden;
-    overflow-y: hidden;
-    position: relative;
-    backface-visibility: hidden;
-    -webkit-overflow-scrolling: auto;
-  }
-
-  .off-canvas-wrapper-inner {
-    @include clearfix;
-    position: relative;
-    width: 100%;
-    min-height: 100%; 
-    transition: transform $offcanvas-transition-length $offcanvas-transition-timing;
-  }
-
-  // Container for page content
-  .off-canvas-content,
-  .#{$maincontent-class} {
-    min-height: 100%;
-    background: $body-background;
-    transition: transform $offcanvas-transition-length $offcanvas-transition-timing;
-    backface-visibility: hidden;
-    z-index: 1;
-    padding-bottom: 0.1px; // Prevents margin collapsing, which would reveal the box shadow of the wrapper
-
-    @if has-value($maincontent-shadow) {
-      box-shadow: $maincontent-shadow;
-    }
+  // Hides overflow on body when an off-canvas panel is open.
+  .is-off-canvas-open {
+    overflow: hidden;
   }
 
-  // Click-to-exit overlay (generated by JavaScript)
-  .js-off-canvas-exit {
-    display: none;
+  // Off-canvas overlay (generated by JavaScript)
+  .js-off-canvas-overlay {
     position: absolute;
     top: 0;
     left: 0;
+
     width: 100%;
     height: 100%;
+
+    transition: opacity $offcanvas-transition-length $offcanvas-transition-timing, visibility $offcanvas-transition-length $offcanvas-transition-timing;
+
     background: $offcanvas-exit-background;
-    cursor: pointer;
-    transition: background $offcanvas-transition-length $offcanvas-transition-timing;
+
+    opacity: 0;
+    visibility: hidden;
+
+    overflow: hidden;
+
+    &.is-visible {
+      opacity: 1;
+      visibility: visible;
+    }
+
+    &.is-closable {
+      cursor: pointer;
+    }
   }
 }
 
-/// Adds basic styles for an off-canvas menu.
-@mixin off-canvas-base {
+// Adds basic styles for an off-canvas wrapper.
+@mixin off-canvas-wrapper() {
+  position: relative;
+  overflow: hidden;
+}
+
+/// Adds basic styles for an off-canvas panel.
+@mixin off-canvas-base(
+  $background: $offcanvas-background,
+  $transition: $offcanvas-transition-length $offcanvas-transition-timing,
+  $fixed: $offcanvas-fixed
+) {
   @include disable-mouse-outline;
-  position: absolute;
-  background: $offcanvas-background;
-  z-index: $offcanvas-zindex;
-  max-height: 100%;
-  overflow-y: auto;
-  transform: translateX(0) translateY(0);
+
+  @if $fixed == true {
+    position: fixed;
+  }
+  @else {
+    position: absolute;
+  }
+
+  z-index: $offcanvas-push-zindex;
+
+  transition: transform $transition;
+  backface-visibility: hidden;
```

Comment:
This I believe it to increase performance in certain browsers/fix flickering: http://blog.teamtreehouse.com/increase-your-sites-performance-with-hardware-accelerated-css

This was in the previous version, but happy to remove if its felt its no longer needed. Also wonder if we should use the `translateZ(0)` method mentioned in that article.